ethstorage/miner/worker.go

```diff
@@ -558,3 +607,7 @@ func (w *worker) getMiningData(t *task, sampleIdx []uint64) ([][]byte, []uint64,
 	}
 	return dataSet, kvIdxs, sampleIdxsInKv, encodingKeys, encodedSamples, nil
 }
+
+func (w *worker) getState() (map[uint64]*MiningState, map[uint64]*SubmissionState) {
+	return w.miningStates, w.submissionStates
```

Comment:
It will be dangerous to return an internal mapping that is accessed by multiple threads, as this can lead to race conditions